l10n_fr_payment_payfip/models/payment_transaction.py

```diff
@@ -0,0 +1,212 @@
+        string="PayFIP state",
+        selection=[
+            ('P', "Effective payment (P)"),
+            ('V', "Effective payment (V)"),
+            ('A', "Abandoned payment (A)"),
+            ('R', "Other cases (R)"),
+            ('Z', "Other cases (Z)"),
+            ('U', "Unknown"),
+        ]
+    )
+
+    payfip_amount = fields.Float(
+        string="PayFIP amount",
+    )
+
+    # endregion
+
+    # region Fields method
+    # endregion
+
+    # region Constrains and Onchange
+    # endregion
+    # region CRUD (overrides)
+    def create(self, vals):
+        res = super(PayFIPTransaction, self).create(vals)
+        if res.acquirer_id.provider == 'payfip':
+            prec = self.env['decimal.precision'].precision_get('Product Price')
+            email = res.partner_email
+            amount = int(float_round(res.amount * 100.0, prec))
+            reference = res.reference.replace('-', ' ')
+            acquirer_reference = '%.15d' % int(uuid.uuid4().int % 899999999999999)
+            res.acquirer_reference = acquirer_reference
+            idop = res.acquirer_id.payfip_get_id_op_from_web_service(email, amount, reference, acquirer_reference)
+            res.payfip_operation_identifier = idop
+
+        return res
+
+    def _get_specific_rendering_values(self, processing_values):
+        res = super()._get_specific_rendering_values(processing_values)
+        if self.provider != 'payfip':
+            return res
+
+        base_url = self.acquirer_id.get_base_url()
+        return {
+            'api_url': PayFIPController._payment_url,
+            'numcli': self.acquirer_id.payfip_customer_number,
+            'exer': fields.Datetime.now().year,
+            'refdet': self.acquirer_reference,
+            'objet': self.reference,
+            'montant': self.amount,
+            'mel': self.partner_email,
+            'urlnotif': urls.url_join("https://tchai.mokatourisme.dev", PayFIPController._notification_url),
+            'urlredirect': urls.url_join("https://tchai.mokatourisme.dev", PayFIPController._return_url),
+            'saisie': 'W' if self.acquirer_id.state == 'prod' else 'T',
+        }
+
+    def _payfip_form_get_tx_from_data(self, idop):
+        if not idop:
+            error_msg = _('PayFIP: received data with missing idop!')
+            _logger.error(error_msg)
+            raise ValidationError(error_msg)
+
+        # find tx -> @TDENOTE use txn_id ?
+        txs = self.env['payment.transaction'].sudo().search([('payfip_operation_identifier', '=', idop)])
+
+        if not txs or len(txs) > 1:
+            error_msg = 'PayFIP: received data for idop %s' % idop
+            if not txs:
+                error_msg += '; no order found'
+            else:
+                error_msg += '; multiple order found'
+            _logger.error(error_msg)
+            raise ValidationError(error_msg)
+        return txs[0]
+
+    def _process_feedback_data(self, idop=False):
```

Comment:
from version 15 you should probably use `self._set_canceled` `self.set_done` and so on instead direct write like here https://github.com/Moka-Tourisme/payment/pull/2/files#diff-bd314b8d14587f90f189f876735fd0d6de4114fe6afef5b653dbcb04f10b97b0R170